assets/react/controllers/Delivery/DeliveryForm.jsx

```diff
@@ -359,11 +357,13 @@ export default function({ storeId, deliveryId, order, isDispatcher }) {
         const { response, error } = await httpClient.post(url, infos)
 
         if (error) {
+          setCalculateResponseData(error.response.data)
           setPriceErrorMessage(error.response.data['hydra:description'])
           setCalculatePrice(0)
         }
 
         if (response) {
+          setCalculateResponseData(response)
```

Comment:
is it needed to have two states here ? (calculateResponseData and calculatePrice)

Reply:
Yes, the difference is that `calculateResponseData` contains not only success but also an error response, which is needed to display debug info when the price is not calculated

Reply:
I mean you can refactor a little bit `setPriceErrorMessage` and the corresponding component so we have only one state in the parent DeliveryForm component ?
but is is a detail no worry